uro98/assignment4/CountIslands.java

```diff
@@ -0,0 +1,57 @@
+
+public class CountIslands {
+
+  /**
+   * Returns the number of islands in a given map.
+   * @param rowCount  the number of rows in the map
+   * @param colCount  the number of columns in the map
+   * @param map       the boolean map where true is land and false is water
+   * @return the number of islands in integer
+   */
+  public static int countIslands(int rowCount, int colCount, boolean[][] map) {
+    int count = 0;
+
+    for (int y = 0; y < colCount; y++) {
+      for (int x = 0; x < rowCount; x++) {
+        if (findIsland(y, x, rowCount, colCount, map)) {
+          count++;
+        }
+      }
+    }
+
+    return count;
+  }
+
+  private static boolean findIsland(int y, int x, int rowCount, int colCount, boolean[][] map) {
+
+    if (isOutOfBounds(y, x, rowCount, colCount)) {
+      return false;
+    }
+
+    if (map[y][x] == true) {
+      // Disable the land as it has been found
+      map[y][x] = false;
```

Comment:
This is very dangerous and should be mentioned in comments of the `countIslands` method. If someone uses `countIslands` in their code they might not expect that their data will get modified.
Imagine that I want to print maps that contain at least 3 islands. I could write code like this:

```
boolean[][] map = getMyAwesomeMap(3, 4);
if (CountIslands.countIslands(3, 4, map) >= 3) {
  printMap(map); // what will be printed here?
}
```